zkml/src/layers/transformer/mha.rs

```diff
@@ -23,22 +27,21 @@ use rayon::iter::{IntoParallelIterator, ParallelIterator};
 use serde::{Deserialize, Serialize};
 
 use crate::{Tensor, layers::provable::LayerOut};
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct MhaQK {
+pub(crate) struct MhaQk {
     num_heads: usize,
     head_dim: usize,
 }
 
-impl MhaQK {
-    pub fn new(num_heads: usize, head_dim: usize) -> Self {
-        Self {
-            num_heads,
-            head_dim,
-        }
-    }
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct MhaFinalMul {
+    num_heads: usize,
+    head_dim: usize,
+    mul: ConcatMatMul,
 }
 
-impl OpInfo for MhaQK {
+impl MhaQk {
```

Comment:
why removing this ? We want still `MhaQK` to be implementing a layer, we dont want to remove the fact it can be inferred or proven... ?

Reply:
Sure it will only be used as a component of MHA but why removing some work that is there anyway ? It didn't seem to hurt to have OpInfo implemented for MhaQK still...